File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/postgresql/DateTimeTypes.kt
package dev.typr.foundationskt.docs.postgresql

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*
import org.postgresql.util.PGInterval
import java.time.Instant
import java.time.LocalDate

@Suppress("unused")
class DateTimeTypes {
    //start
    val dateType: PgType<LocalDate> = PgTypes.date
    val timestamptzType: PgType<Instant> = PgTypes.timestamptz
    val intervalType: PgType<PGInterval> = PgTypes.interval
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/postgresql/DateTimeTypes.scala
package dev.typr.foundationssc.docs.postgresql
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

import org.postgresql.util.PGInterval
import java.time.{Instant, LocalDate}

@SuppressWarnings(Array("unused"))
object DateTimeTypes:
  // start
  val dateType: PgType[LocalDate] = PgTypes.date
  val timestamptzType: PgType[Instant] = PgTypes.timestamptz
  val intervalType: PgType[PGInterval] = PgTypes.interval
  // stop
